rpc/service.py

```diff
@@ -367,10 +368,62 @@ async def RegisterTaskGraph(self, request, context):
                 num_executors=0,
             )
 
+        self._logger.info(
+            "Attempting to register application ID %s with name %s",
+            request.id,
+            request.name,
+        )
+        # Check if query is from TPC-H workload.
+        # If yes, retrieve profiled slots and runtime info. If no, use default values
+        is_tpch_query = False
+        tpch_query_all_stage_info = None
+        if request.name.startswith("TPCH_"):
+            is_tpch_query = True
+            # retrieve tasks-per-stage and runtime info based on query number
+            tpch_query_num = request.name.split("TPCH_Q", 1)[1]
+            tpch_query_all_stage_info = get_all_stage_info_for_query(tpch_query_num)
+            same_structure, stage_id_mapping = verify_and_relable_tpch_app_graph(
+                query_num=tpch_query_num, dependencies=request.dependencies
+            )
+
+            # return failure message if not tpch app isnt of same DAG structure
+            if not same_structure:
+                self._logger.warning(
+                    "TPCH application with ID %s and name %s couldn't be registered."
+                    "DAG structure mismatch!",
+                    request.id,
+                    request.name,
+                )
+                return erdos_scheduler_pb2.RegisterTaskGraphResponse(
+                    success=False,
+                    message=f"TPCH application ID {request.id} with name {request.name}"
+                    f" couldn't be registered. DAG structure mismatch!",
+                    num_executors=0,
+                )
+
         # Construct all the Tasks for the TaskGraph.
         task_ids_to_task: Mapping[int, Task] = {}
+        default_resource = Resources(
+            resource_vector={Resource(name="Slot_CPU", _id="any"): 30}
+        )
+        default_runtime = EventTime(20, EventTime.Unit.US)
```

Comment:
```suggestion
        default_runtime = EventTime(20, EventTime.Unit.S)
```

Do you have any runtimes that are lower than a second?